framework/src/modules/http_api/controllers/blocks.js

```diff
@@ -108,6 +111,67 @@ BlocksController.getBlocks = function(context, next) {
 	});
 };
 
+/**
+ * Gets address by public.
+ *
+ * @private
+ * @param {publicKey} publicKey Public key
+ * @returns {address} address
+ * @todo Replace by Lisk Elements once integrated.
+ */
+__private.getAddressByPublicKey = function(publicKey) {
+	const publicKeyHash = crypto
+		.createHash('sha256')
+		.update(publicKey, 'hex')
+		.digest();
+	const temp = Buffer.alloc(8);
+
+	for (let i = 0; i < 8; i++) {
+		temp[i] = publicKeyHash[7 - i];
+	}
+
+	const address = `${Bignum.fromBuffer(temp).toString()}L`;
+	return address;
+};
+
+/**
+ * Parse raw block data from database into expected API response type for blocks
+ *
+ * @param {Object} raw Raw block data from database
+ * @return {block} Block formatted according to API specification
+ */
+function parseBlockFromDatabase(raw) {
+	if (!raw.id) {
+		return null;
+	}
+
+	const block = {
+		id: raw.id,
+		version: parseInt(raw.version),
+		timestamp: parseInt(raw.timestamp),
+		height: parseInt(raw.height),
+		previousBlock: raw.previousBlockId,
+		numberOfTransactions: parseInt(raw.numberOfTransactions),
+		totalAmount: new Bignum(raw.totalAmount),
+		totalFee: new Bignum(raw.totalFee),
+		reward: new Bignum(raw.reward),
+		payloadLength: parseInt(raw.payloadLength),
+		payloadHash: raw.payloadHash,
+		generatorPublicKey: raw.generatorPublicKey,
+		generatorId: __private.getAddressByPublicKey(raw.generatorPublicKey),
+		blockSignature: raw.blockSignature,
+		confirmations: parseInt(raw.confirmations),
+	};
+
+	if (raw.transactions) {
+		block.transactions = raw.transactions;
+	}
+
+	block.totalForged = block.totalFee.plus(block.reward).toString();
```

Comment:
why not do this when creating the `block` above?

Reply:
Perhaps we could add unit tests for this method.